### ISSUE TITLE
Training a Pertained Model 

### ISSUE THREAD

**Issue by tabarkarajab** [ultralytics/yolov5#3949]
**I need to ask you another question that I have trained a model using this PyTorch custom dataset, initially, I used the yolov5s.pt file for the weights, but now I have trained a model, need to increase the dataset, so do I need to retrain on yolov5s.pt file or I used mine best.pt file?**

**Comment by glenn-jocher** [comment]
@tabarkarajab with YOLOv5 you can start training any dataset from any pretrained weights. Your results will vary by the pretrained weights you start from (or if you start from scratch). You can experiment with starting from the official COCO-trained weights vs starting from your own custom-dataset pretrained weights, i.e.:
```bash
python train.py --data data.yaml --weights yolov5s.pt  # start from official COCO-trained YOLOv5s
python train.py --data data.yaml --weights path/to/best.pt  # start from your own data-trained YOLOv5s
```

**Comment by tabarkarajab** [comment]
Thanks, @glenn-jocher  But what I get is this! can you please tell me my data is overfitted on unfitted? https://wandb.ai/tabarka/YOLOv5/runs/5ea4xaid?workspace=user-tabarka

**Comment by glenn-jocher** [comment]
@tabarkarajab it's very simple. If your validation losses have not started increasing then your data has not started overfitting, and you need to train more.

**Comment by tabarkarajab** [comment]
means need to increase my dataset although I'm using 1306 images

**Comment by tabarkarajab** [comment]
hi @glenn-jocher, sorry for teasing you again, kindly can you check these training results as I'm not getting results as per my testing video. 

https://wandb.ai/tabarka/YOLOv5/runs/29lkzx2t?workspace=user-tabarka

**Comment by tabarkarajab** [comment]
my trained model is detecting the unglove dataset even if I'm giving the glove video.

**Comment by glenn-jocher** [comment]
@tabarkarajab 👋 Hello! Thanks for asking about improving YOLOv5 🚀 training results. 

Most of the time good results can be obtained with no changes to the models or training settings, **provided your dataset is sufficiently large and well labelled**. If at first you don't get good results, there are steps you might be able to take to improve, but we always recommend users **first train with all default settings** before considering any changes. This helps establish a performance baseline and spot areas for improvement.

If you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

We've put together a full guide for users looking to get the best results on their YOLOv5 trainings below.

## Dataset

- **Images per class.** ≥ 1500 images per class recommended
- **Instances per class.** ≥ 10000 instances (labeled objects) per class recommended
- **Image variety.** Must be representative of deployed environment. For real-world use cases we recommend images from different times of day, different seasons, different weather, different lighting, different angles, different sources (scraped online, collected locally, different cameras) etc.
- **Label consistency.** All instances of all classes in all images must be labelled. Partial labelling will not work.
- **Label accuracy.** Labels must closely enclose each object. No space should exist between an object and it's bounding box. No objects should be missing a label.
- **Background images.** Background images are images with no objects that are added to a dataset to reduce False Positives (FP). We recommend about 0-10% background images to help reduce FPs (COCO has 1000 background images for reference, 1% of the total). No labels are required for background images.

<a href="https://arxiv.org/abs/1405.0312"><img width="800" src="https://user-images.githubusercontent.com/26833433/109398377-82b0ac00-78f1-11eb-9c76-cc7820669d0d.png" alt="COCO Analysis"></a>


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. VOC, VisDrone, GlobalWheat). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
                                             custom_pretrained.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. COCO, Objects365, OIv6). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```


## Training Settings

Before modifying anything, **first train with default settings to establish a performance baseline**. A full list of train.py settings can be found in the [train.py](https://github.com/ultralytics/yolov5/blob/master/train.py) argparser.

- **Epochs.** Start with 300 epochs. If this overfits early then you can reduce epochs. If overfitting does not occur after 300 epochs, train longer, i.e. 600, 1200 etc epochs.
- **Image size.** COCO trains at native resolution of `--img 640`, though due to the high amount of small objects in the dataset it can benefit from training at higher resolutions such as `--img 1280`. If there are many small objects then custom datasets will benefit from training at native or higher resolution. Best inference results are obtained at the same `--img` as the training was run at, i.e. if you train at `--img 1280` you should also test and detect at `--img 1280`.
- **Batch size.** Use the largest `--batch-size` that your hardware allows for. Small batch sizes produce poor batchnorm statistics and should be avoided.
- **Hyperparameters.** Default hyperparameters are in [hyp.scratch.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyp.scratch.yaml). We recommend you train with default hyperparameters first before thinking of modifying any. In general, increasing augmentation hyperparameters will reduce and delay overfitting, allowing for longer trainings and higher final mAP. Reduction in loss component gain hyperparameters like `hyp['obj']` will help reduce overfitting in those specific loss components. For an automated method of optimizing these hyperparameters, see our [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).

## Further Reading

If you'd like to know more a good place to start is Karpathy's 'Recipe for Training Neural Networks', which has great ideas for training that apply broadly across all ML domains:
http://karpathy.github.io/2019/04/25/recipe/

**Comment by tabarkarajab** [comment]
Hi, what we called this X and Y-axis in the results of the training?


![media_images_Results_100_0](https://user-images.githubusercontent.com/49900380/136692713-588aa058-cb07-488c-a7e6-9c406a7f6e1e.png)

**Comment by glenn-jocher** [comment]
@tabarkarajab x axis is epoch, y axis is shown in title

**Comment by tabarkarajab** [comment]
thanks, @glenn-jocher, I have another question, I have trained my model but it has reduced the size to around 54MB which is causing the delay in detection of the object on jetson nano, is there any way I can do it?

**Comment by glenn-jocher** [comment]
@tabarkarajab YOLOv5 🚀 training checkpoints should be 4X the size of final trained checkpoints, as they carry not just a FP16 model, but a FP16 [EMA](https://www.tensorflow.org/api_docs/python/tf/train/ExponentialMovingAverage) and an FP32 optimizer of the same size as the model (each model parameter has its own FP32 gradient saved within the optimizer). Final checkpoints contain only an FP16 model, with the EMA and optimizer both stripped after the final epoch of training:
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/train.py#L435-L439

The `strip_optimizer()` function updates the checkpoint dictionary value for `model`, replacing it with `ema`, and sets `ema` and `optimizer` keys to `None`, which will reduce the checkpoint size by 3/4.
https://github.com/ultralytics/yolov5/blob/75feeb797c4a9553f4274860c6c278f1fc628f60/utils/general.py#L495-L509

Final trained checkpoints after `strip_optimizer()` should match the README table sizes below.
<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>
